rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -86,11 +92,13 @@ static size_t calculateMaxAlign(const rosidl_typesupport_introspection_cpp::Mess
                     alignment = alignof(uint64_t);
                     break;
                 case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_STRING:
+                    // StringHelper exposes rosidl_generator_c__String as std::string
+                    // so we don't need differentiate between C and C++ introspection typesupport
                     alignment = alignof(std::string);
                     break;
                 case ::rosidl_typesupport_introspection_cpp::ROS_TYPE_MESSAGE:
                     {
-                        const ::rosidl_typesupport_introspection_cpp::MessageMembers *sub_members = (const ::rosidl_typesupport_introspection_cpp::MessageMembers*)member.members_->data;
+                        const MembersType *sub_members = (const MembersType*)member.members_->data;
```

Comment:
@dirk-thomas given that this repository is owned by eProsima (who may have their own codestyle guidelines, different from ROS2's), and that the changes are significant already, I tried to abstain from making further changes in order to help the reviewing process. However, if you think the code needs any changes, feel free to make them. Thanks!